.github/workflows/e2e.yaml

```diff
@@ -35,10 +35,28 @@ jobs:
               echo "Using tag $tag"
               echo "simd-tag=$tag" >> $GITHUB_OUTPUT
             fi
+
+  build-e2e:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+      - name: Build e2e
+        run: |
+          cd e2e
+          test_dirs="$(ls -A tests)"
+          for td in $test_dirs
+          do
+            go test -c "./tests/${td}"
+          done
```

Comment:
added this job to enable the tests to fail earlier with build errors. Before this, the tests ran and only failed after the e2e job began
.